zmq.hpp

```diff
@@ -1020,7 +1020,9 @@ namespace zmq
             return *this;
         }
 
-        void add (zmq::socket_t &socket, short events, std::function<void(void)> &handler)
+        using handler_t = std::function<void(short)>;
+
+        void add (zmq::socket_t &socket, short events, handler_t &handler)
```

Comment:
This should be done in a backwards-compatible way, i.e. keep a function with the old signature that wraps a std:: function<void(void)> and passes it to the function with the new signature.

Reply:
thank you for review. this is draft so no need for that.

Reply:
Right. Since it is related to a libzmq draft api, it can be regarded draft in cppzmq as well.